### PR TITLE
Port cufflinks

### PR DESCRIPTION
**Pull request by nicolaskruchten**
New version of https://plotly.com/python/v3/ipython-notebooks/cufflinks/

### CODE REVIEW

**Comment by emmanuelle** [comment]
CI fails with 
```
Checking... are there posts with order > 5 and 'page_type: example_index'?
NOT PASSED!
```
for the new tutorial

**Comment by emmanuelle** [comment]
did you merge my recent chart-studio fixes and is it something different for the 2.7 CI failure?

**Comment by nicolaskruchten** [comment]
This is in doc-prod so your fixes have not been applied yet, no, not until the next release.

File: doc/python/cufflinks.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+
+> Issues and questions regarding Cufflinks should be [raised in the Cufflinks repository](https://github.com/santosjorge/cufflinks/issues/new).
+
+```python
+import cufflinks as cf
+import pandas as pd
+import numpy as np
+
+df = pd.DataFrame(np.random.randn(1000, 2), columns=['A', 'B']).cumsum()
+fig = df.iplot(asFigure=True, xTitle="The X Axis", yTitle="The Y Axis", title="The Figure Title")
```
Comment (emmanuelle) [inline doc/python/cufflinks.md]:
```suggestion
fig = df.iplot(asFigure=True, xTitle="The X Axis",
                    yTitle="The Y Axis", title="The Figure Title")
```

Comment (emmanuelle) [reply doc/python/cufflinks.md]: avoid horizontal scrolling

File: doc/python/cufflinks.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+fig.show()
+```
+
+### Line and Box Annotations
+
+```python
+import cufflinks as cf
+
+df=cf.datagen.lines(4)
+fig = df.iplot(asFigure=True, hline=[2,4],vline=['2015-02-10'])
```
Comment (emmanuelle) [inline doc/python/cufflinks.md]:
```suggestion
fig = df.iplot(asFigure=True, hline=[2,4], vline=['2015-02-10'])
```

**Comment by emmanuelle** [comment]
Looks good! It could be nice to add an example calling `fig.update_layout` or `update_traces` after calling `iplot` but this can be done later so :dancer:

**Comment by nicolaskruchten** [comment]
Thanks!

**Pull request merged**